src/main/java/org/elasticsearch/index/query/SpanTermQueryBuilder.java

```diff
@@ -19,83 +19,95 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.spans.SpanTermQuery;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.lucene.BytesRefs;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.mapper.MapperService;
 
 import java.io.IOException;
 
 /**
- *
+ * A Span Query that matches documents containing a term.
+ * @see SpanTermQuery
  */
-public class SpanTermQueryBuilder extends BaseQueryBuilder implements SpanQueryBuilder, BoostableQueryBuilder<SpanTermQueryBuilder> {
-
-    private final String name;
-
-    private final Object value;
-
-    private float boost = -1;
-
-    private String queryName;
-
+public class SpanTermQueryBuilder extends BaseTermQueryBuilder implements SpanQueryBuilder, BoostableQueryBuilder<SpanTermQueryBuilder> {
+    /** {@inheritDoc} */
     public SpanTermQueryBuilder(String name, String value) {
-        this(name, (Object) value);
+        super(name, (Object) value);
     }
-
+    /** {@inheritDoc} */
     public SpanTermQueryBuilder(String name, int value) {
-        this(name, (Object) value);
+        super(name, (Object) value);
     }
-
+    /** {@inheritDoc} */
     public SpanTermQueryBuilder(String name, long value) {
-        this(name, (Object) value);
+        super(name, (Object) value);
     }
-
+    /** {@inheritDoc} */
     public SpanTermQueryBuilder(String name, float value) {
-        this(name, (Object) value);
+        super(name, (Object) value);
     }
-
+    /** {@inheritDoc} */
     public SpanTermQueryBuilder(String name, double value) {
-        this(name, (Object) value);
+        super(name, (Object) value);
+    }
+    /** {@inheritDoc} */
+    public SpanTermQueryBuilder(String name, Object value) {
+        super(name, value);
     }
 
-    private SpanTermQueryBuilder(String name, Object value) {
-        this.name = name;
-        this.value = value;
+    SpanTermQueryBuilder() {
+        super();
+        // for testing and serialisation only
     }
 
+    /** @see org.apache.lucene.search.Query#getBoost() */
     @Override
     public SpanTermQueryBuilder boost(float boost) {
         this.boost = boost;
         return this;
     }
-
-    /**
-     * Sets the query name for the filter that can be used when searching for matched_filters per hit.
-     */
+    
+    /** Sets the query name for the filter that can be used when searching for matched_filters per hit. */
     public SpanTermQueryBuilder queryName(String queryName) {
         this.queryName = queryName;
         return this;
     }
 
+    /** {@inheritDoc} */
     @Override
     public void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(SpanTermQueryParser.NAME);
```

Comment:
Not sure, but if we use parserName() instead of the constant here, can we move the whole doXContent method do the BaseTermQueryBuilder?